### PR TITLE
Add unit tests for core logic

### PR DESCRIPTION
**Pull request by 0x80**
Add 74 unit tests across 4 test files covering the core pure functions:

- Casing functions (kebab/snake conversion with edge cases like acronyms, consecutive capitals)
- File and segment conversion (preserving underscore-prefixed files, all-uppercase files, Next.js route params, case-insensitive filesystem handling with __tmp suffix)
- Path transformation (all target prefixes, nested relative paths, extension preservation)
- AST codemod (ES6 imports, named/all exports, TypeScript import type, require(), new URL(), mixed imports)

Also exports `getNewPathPhaseOne`, `getNewPathPhaseTwo`, and `transformPath` so they can be tested directly.

Note: tests document that jscodeshift's tsx parser doesn't match `ImportExpression` for dynamic `import()` calls, making those handlers in `update-imports-and-exports.ts` dead code.

Scope: tests
Visibility: internal

### CODE REVIEW

File: src/codemods/helpers/update-source-value.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function getUpdatedSource<T extends ASTNode>(
   return null;
 }
 
-function transformPath(
+export function transformPath(
   filePath: string,
   casingFn: (str: string) => string,
 ): string {
```
[the remark this answers is not in the document]
Comment (0x80) [reply src/codemods/helpers/update-source-value.ts:29]: Valid observation about the source code, but out of scope for this test PR. The codemod runs after both rename phases complete (phase 2 strips `__tmp`), so in practice the import specifiers produced by `transformPath` during the codemod run won't contain `__tmp` — the codemod operates on the already-renamed files. If this is worth addressing, it should be a separate issue.

File: src/codemods/helpers/update-imports-and-exports.test.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  const root = j(source);
+  return updateImportsAndExports(j, root, toKebab);
+}
+
+describe("updateImportsAndExports", () => {
+  it("transforms ES6 import declarations", () => {
+    const input = `import { Button } from './MyComponent';`;
+    const output = transform(input);
+    expect(output).toContain(`"./my-component"`);
+  });
```
[the remark this answers is not in the document]
Comment (0x80) [reply src/codemods/helpers/update-imports-and-exports.test.ts:19]: Intentional — pinning the current output behavior catches unexpected changes in jscodeshift/recast versions. If the quoting behavior changes, we want the tests to flag it.

File: src/codemods/helpers/update-imports-and-exports.test.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    expect(output).toContain(`"./my-worker"`);
+  });
+
+  it("does not transform dynamic import() (tsx parser treats import() as CallExpression, not ImportExpression)", () => {
+    // jscodeshift's tsx parser parses import() as CallExpression with callee.type "Import"
+    // rather than ImportExpression, so j.ImportExpression doesn't match
+    const input = `async function load() { const mod = await import('./MyModule'); }`;
+    const output = transform(input);
+    expect(output).toBeNull();
+  });
```
[the remark this answers is not in the document]
Comment (0x80) [reply src/codemods/helpers/update-imports-and-exports.test.ts:63]: Fixing the `ImportExpression` handling is a source code change, out of scope for this test PR. The tests document the current behavior. A follow-up PR can address this.

File: src/lib/convert-files.test.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  it("adds __tmp suffix to segments that only differ in casing", () => {
+    expect(getNewPathPhaseOne("src/Mydir/file.ts", toKebab)).toBe(
+      "src/mydir__tmp/file.ts",
+    );
+  });
+
+  it("preserves segments that don't need conversion", () => {
+    expect(getNewPathPhaseOne("src/lib/config.ts", toKebab)).toBe(
+      "src/lib/config.ts",
+    );
```
[the remark this answers is not in the document]
Comment (0x80) [reply src/lib/convert-files.test.ts:106]: This tool targets macOS/Linux JS/TS codebases. Windows support is not a goal, so POSIX paths in tests are fine.
